### ISSUE TITLE
wrong indent when open text file including CJK characters

### ISSUE THREAD

**Issue by zombie110year** [nushell/nushell#1784]
# Description

When using `open` command to read a text file which includes CJK character,
each line will add a wrong indent.

# To Reproduce

1. edit a text includes CJK character, and named with random filename, e.g. *text.txt*
2. run command `open text.txt`
3. view the output:

```
# A Example file to show nu's bug

This is a common line didn't includes CJK char.
and then, the next line has a same indent level as uppon.

但是，这一样是全汉字的，
            所以第二行就出现了错误的缩进。                                                                                                         Although the next line is full of ascii, the indent remains.

                           if a line is not full 汉字, the indent is still been added.
                             just like this.

                             And the, I found that, every 汉字 will add a common space.
                               This line addes 2 spaces than previous line, because of 2 CJK char.
```

# **Expected behavior**

the raw content of file is:

```
# A Example file to show nu's bug

This is a common line didn't includes CJK char.
and then, the next line has a same indent level as uppon.

但是，这一样是全汉字的，
所以第二行就出现了错误的缩进。
Although the next line is full of ascii, the indent remains.

if a line is not full 汉字, the indent is still been added.
just like this.

And the, I found that, every 汉字 will add a common space.
This line addes 2 spaces than previous line, because of 2 CJK char.
```

# Configuration
- OS: Windows
- Version: 0.14.0

# Others

I guess it caused by rendering full-width characters.

**Comment by thegedge** [comment]
The problem is here: https://github.com/nushell/nushell/blob/3144dc7f9375b9faff9848529eeb2c2df1aed586/crates/nu_plugin_textview/src/textview.rs#L68-L73

We're assuming a 1:1 mapping from unicode char to terminal column. What we probably need to be doing is iterating over grapheme clusters.

**Comment by jonathandturner** [comment]
I haven't tried it, but this looks promising: https://crates.io/crates/unicode-width

**Comment by shaaraddalvi** [comment]
I can give it a try if that's fine! :smile:

**Comment by jonathandturner** [comment]
@shaaraddalvi - yup, I'll assign it to you

**Comment by onthebridgetonowhere** [comment]
I just tried now on my Windows machine and the behavior was as expected. Seems to have been fixed - maybe by #2092? Shall we close this issue for now?

```
   1   │ # A Example file to show nu's bug
   2   │
   3   │ This is a common line didn't includes CJK char.
   4   │ and then, the next line has a same indent level as uppon.
   5   │
   6   │ 但是，这一样是全汉字的，
   7   │ 所以第二行就出现了错误的缩进。
   8   │ Although the next line is full of ascii, the indent remains.
   9   │
  10   │ if a line is not full 汉字, the indent is still been added.
  11   │ just like this.
  12   │ 
  13   │ And the, I found that, every 汉字 will add a common space.
  14   │ This line addes 2 spaces than previous line, because of 2 CJK char.
```
